source/numerics/data_out_dof_data.cc

```diff
@@ -546,7 +585,122 @@ namespace internal
       return get_vector_element(*vector, cell_number);
     }
 
+    // FIXME:
+    // wrap the actual assignment into a function in order to have a
+    // convenient way of explicit instantiation for complex-value case,
+    // which is not supported yet. An alternative would have been to
+    // explicit instantiate the whole (!) DataEntry class, which is too much
+    // code repetition for a temporary workaround.
+    // Remove after DataOut is extended to complex algebra.
+    namespace
+    {
+      template <typename number>
+      void
+      write_patch_values (const std::vector<number> &tmp,
+                          std::vector<double> &patch_values)
+      {
+        for (unsigned int i = 0; i < tmp.size(); i++)
+          patch_values[i] = tmp[i];
```

Comment:
It's not wrong to do it this way, but you could make the code one line shorter by using `std::copy` ;-)